helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/PartitionMovementConstraint.java

```diff
@@ -71,48 +43,13 @@ protected double getAssignmentScore(AssignableNode node, AssignableReplica repli
     String state = replica.getReplicaState();
 
     if (bestPossibleAssignment.isEmpty()) {
-      // If bestPossibleAssignment of the replica is empty, indicating this is a new replica.
```

Comment:
I think we should keep this comment. Or it would be hard to explain why we are using baselineAssignment as the input here.

Reply:
I have already added that to the javadoc of this constraint. I can reiterate it, sure.

Reply:
Yeah, but it is OK just leave it here since it is still valid.